lib/src/spec.rs

```diff
@@ -89,6 +89,22 @@ pub struct ImageReference {
     pub signature: Option<ImageSignature>,
 }
 
+impl ImageReference {
+    /// Returns a canonicalized version of this image reference, preferring the digest over the tag if both are present.
+    pub fn canonicalize(&self) -> ImageReference {
```

Comment:
I think this should take `self` not `&self` and then the no-op case is a passthrough and not a clone.

Reply:
Done!  
FYI where we're calling this function, we're cloning the return value anyway.